packages/react-instantsearch/src/core/createConnector.js

```diff
@@ -63,24 +63,37 @@ export default function createConnector(connectorDesc) {
 
         const { ais: { store, widgetsManager } } = context;
         const canRender = false;
+
+        const initialUiState = connectorDesc.initialUiState
+          ? connectorDesc.initialUiState(props)
+          : {};
+
         this.state = {
-          props: this.getProvidedProps({ ...props, canRender }),
+          uiState: initialUiState,
+          props: this.getProvidedProps({
+            ...props,
+            ...initialUiState,
```

Comment:
I don't like that, we should not merge that in the `props`. We should pass the `uiState` as an argument to the connector definition.